services/libs/tracing/src/tracer.ts

```diff
@@ -0,0 +1,80 @@
+import opentelemetry, { Tracer } from '@opentelemetry/api'
+import { NodeSDK } from '@opentelemetry/sdk-node'
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-grpc'
+import { Resource, ResourceAttributes } from '@opentelemetry/resources'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+
+import {
+  awsEc2Detector,
+  awsEcsDetector,
+  awsEksDetector,
+} from '@opentelemetry/resource-detector-aws'
+
+import { BunyanInstrumentation } from '@opentelemetry/instrumentation-bunyan'
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
+import { ExpressInstrumentation } from '@opentelemetry/instrumentation-express'
+import { AwsInstrumentation } from '@opentelemetry/instrumentation-aws-sdk'
+import { RedisInstrumentation } from '@opentelemetry/instrumentation-redis'
+import { SequelizeInstrumentation } from 'opentelemetry-instrumentation-sequelize'
+
+let sdk: NodeSDK | undefined
+let isInitialized = false
+
+export const getServiceTracer = (): Tracer => {
+  const service = process.env['SERVICE']
+  if (!service) {
+    throw new Error('Environment variable `SERVICE` is not set')
+  }
+
+  if (isInitialized) {
+    return opentelemetry.trace.getTracer(service)
+  }
+
+  const attrs: ResourceAttributes = {
+    [SemanticResourceAttributes.SERVICE_NAME]: service,
+  }
+
+  // Apply specific resource attributes if process is running in Kubernetes.
+  // This is not handled by AWS resource detectors.
+  if (process.env['KUBE_MODE'] == '1') {
+    attrs[SemanticResourceAttributes.K8S_POD_NAME] = process.env['HOSTNAME']
+  }
+
+  sdk = new NodeSDK({
```

Comment:
How do we pass OTEL URL and other configuration options here? I would prefer to have it explicit vs. some AWS autoconfiguration happening in the SDK.